net/inet/ipv6_setsockopt.c

```diff
@@ -86,28 +86,17 @@ int ipv6_setsockopt(FAR struct socket *psock, int option,
       /* Handle MLD-related socket options */
 
       case IPV6_JOIN_GROUP:       /* Join a multicast group */
-        {
-          FAR const struct ipv6_mreq *mrec ;
-
-          mrec = (FAR const struct ipv6_mreq *)value;
-          ret = mld_joingroup(mrec);
-        }
+        ret = mld_joingroup(value);
```

Comment:
I think you still need to use the casting, because some compiler could raise some warnings.
```
ret = mld_joingroup((FAR const struct ipv6_mreq *) value);
```

Reply:
I'm really not sure about this. I think cast from/to `void *` do not require additional type cast

Reply:
Both C/C++ allow any pointer type convert to void * without cast, but only C allow void * convert to other pointer type without cast.